py-polars/polars/internals/anonymous_scan.py

```diff
@@ -52,6 +52,9 @@ def _scan_pyarrow_dataset_impl(
     """
     _filter = None
     if predicate:
+        # imports are used by inline python evaluated by `eval`
+        from polars.utils import _to_python_time, _to_python_datetime, _to_python_timedelta  # noqa:
```

Comment:
@alexander-beedie @stinodego any idea how I can stop ruff from removing these imports?

Reply:
I think you can just add `# noqa: F401` , right?

Reply:
No, it still removes an import line. :/

Reply:
Is there a way to turn this autofix off completely for now?

Reply:
You can remove this line:
https://github.com/pola-rs/polars/blob/579de59a3df321a842a79f652f2b69fd35142f68/py-polars/pyproject.toml#L107

Though I'd rather keep that - you can set error codes as 'unfixable' by adding something like this to the config:
```toml
unfixable = ["F401"]
```

I have configured my VSCode to disable autofixing for a few lints:

```
    "ruff.importStrategy": "fromEnvironment",
    "ruff.args": [
        "--unfixable",
        "F841,F401"
    ],
```

Reply:
Thanks I will try that.

Reply:
Ok, it was due to the black formatting to multiline that the `noqa` didn't work. Fixed it now. Thanks for the help.  :)

Reply:
It can also be due an older ruff version. Recent versions at least don't remove `import pyarrow.csv` imports anymore after a `import pyarrow as pa`.